it/LoginISpec.scala

```diff
@@ -0,0 +1,266 @@
+            summaryErrorMessages = Seq(
+              ExpectedSummaryErrorMessage("Enter a username", "username"),
+              ExpectedSummaryErrorMessage("Enter a password", "password")
+            ),
+            errorMessageForUsername = Some("Enter a username"),
+            errorMessageForPassword = Some("Enter a password")
+          )
+        )
+
+      }
+      "username is incorrect" in {
+
+        val values = Map("username" -> "1230", "password" -> correctPassword)
+
+        val response = submitFromLoginPage(values)
+
+        assertPage(
+          response,
+          BAD_REQUEST,
+          ExpectedLoginPage(
+            title = "Sign in",
+            headerTitle = "Pan-Archival Catalogue",
+            legend = "Sign in: Pan-Archival Catalogue",
+            logoutLinkVisible = false,
+            summaryErrorTitle = Some("There is a problem"),
+            summaryErrorMessages = Seq(
+              ExpectedSummaryErrorMessage("Username and/or password is incorrect.", "")
+            )
+          )
+        )
+
+      }
+      "password is incorrect" in {
+
+        val values = Map("username" -> correctUsername, "password" -> "1230")
+
+        val response = submitFromLoginPage(values)
+
+        assertPage(
+          response,
+          BAD_REQUEST,
+          ExpectedLoginPage(
+            title = "Sign in",
+            headerTitle = "Pan-Archival Catalogue",
+            legend = "Sign in: Pan-Archival Catalogue",
+            logoutLinkVisible = false,
+            summaryErrorTitle = Some("There is a problem"),
+            summaryErrorMessages = Seq(
+              ExpectedSummaryErrorMessage("Username and/or password is incorrect.", "")
+            )
+          )
+        )
+
+      }
+    }
+
+    "succeed" when {
+      "username and password are correct" in {
+
+        val values = Map("username" -> correctUsername, "password" -> correctPassword)
+
+        val responseForLoginSubmission = submitFromLoginPage(values)
+
+        assertRedirection(responseForLoginSubmission, "/edit-set/1")
+
+        val responseForEditSetPage = getEditSetPage(getSessionCookie(responseForLoginSubmission))
+
+        assertPage(
+          responseForEditSetPage,
+          OK,
+          ExpectedEditSetPage(
+            title = "Edit set",
+            logoutLinkVisible = true
+          )
+        )
+
+      }
+    }
+  }
+
+  private def submitFromLoginPage(values: Map[String, String]): WSResponse = {
+    val securityInfo = getSecurityInfoFromLoginPage
+    await(
+      wsUrl("/login")
+        .withFollowRedirects(false)
+        .addCookies(securityInfo.sessionCookie)
+        .post(values ++ Map("csrfToken" -> securityInfo.csrfToken))
+    )
+  }
+
+  private def getEditSetPage(sessionCookie: WSCookie): WSResponse =
+    await(
+      wsUrl("/edit-set/1")
+        .withFollowRedirects(false)
+        .addCookies(sessionCookie)
+        .get()
+    )
+
+  private def getLoginPage: WSResponse =
```

Comment:
I  would add empty parens "()" after this method, to make it clear that it's side-effecting, see https://docs.scala-lang.org/style/method-invocation.html

Reply:
Sure. But what's the side-effect in this case?

Reply:
The HTTP client is making a GET request to the server - calling it two times may result in different values, for example if the HTTP server is not responding.

Reply:
Ah OK - in a functional sense, yes.

I took getting the data from the server as the main effect.

But I'll change it.

Reply:
Changed.